### DISCUSSION TITLE
Intergate Self Access-Token Service for every collection Restful-API

### DISCUSSION THREAD

**Discussion by imiskolee** [directus/directus#12373] [Archive]
hi,team:

If we have a existing web system and token-based permission verify function, I am thinking a way to integrate it's on this platform.

* add a token string to request header
* directus call our service to verify token and get token info(user_id), Inject it to current API context.
* focus apply the user_id to collection query `SQL(select * from my_collection where user_id = ${REQ.user_id})`

how to let it works?

**Comment by rijkvanzanten** [comment]
You can use the `authenticate` filter header to add your custom token parsing 👍🏻 

https://docs.directus.io/extensions/hooks/#filter-events
